Regression failure for tests/ert_tests/data/test_measured_data.py

**Describe the bug**
There is a test regression occurring for the test:
tests/ert_tests/data/test_measured_data.py



**To Reproduce**
Steps to reproduce the behavior:
1. `git clone git@github.com:equinor/ert.git`
2. `cd ert`
3. `pip install -e .`
4. `pip install -r dev-requirements.txt`
5. `pytest tests/ert_tests/data/test_measured_data.py`

**Expected behavior**
Tests should pass

**Actual behavior**
Tests failures
 `platform linux -- Python 3.9.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
PyQt5 5.15.6 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/thannant/development/equinor/ert-testting, configfile: pyproject.toml
plugins: cov-3.0.0, anyio-3.5.0, qt-4.0.2, asyncio-0.17.2, snapshot-0.8.0, subtests-0.6.0, timeout-2.1.0, benchmark-3.4.1, hypothesis-5.49.0, ert-storage-0.3.7, schemathesis-1.3.4, flaky-3.7.0
asyncio: mode=legacy
collected 22 items                                                                                               

tests/ert_tests/data/test_measured_data.py ....FF................                                          [100%]

==================================================== FAILURES ====================================================
_______________________ test_remove_failed_realizations[input_dataframe1-expected_result1] _______________________

input_dataframe = key_index      0     1     2
data_index     0     1     2
1           None  None  None
expected_result =            obs_key          
key_index        0    1    2
data_index       0    1    2
OBS            1.0  2.0  3.0
STD            4.0  5.0  6.0
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f148edd1e20>, facade = <Mock id='139726272587616'>
measured_data_setup = <function measured_data_setup.<locals>._setup at 0x7f148ee5dd30>
valid_obs_data = key_index     0    1    2
data_index    0    1    2
OBS         1.0  2.0  3.0
STD         4.0  5.0  6.0

    @pytest.mark.parametrize(
        "input_dataframe,expected_result",
        [
            (
                pd.DataFrame(data=[[7, 8, 9]], index=[1]),
                pd.DataFrame(
                    data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]],
                    index=["OBS", "STD", 1],
                ),
            ),
            (
                pd.DataFrame(data=[[None, None, None]], index=[1]),
                pd.DataFrame(data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]], index=["OBS", "STD"]),
            ),
            (
                pd.DataFrame(data=[[7, 8, None]], index=[1]),
                pd.DataFrame(
                    data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, None]],
                    index=["OBS", "STD", 1],
                ),
            ),
        ],
    )
    def test_remove_failed_realizations(
        input_dataframe,
        expected_result,
        monkeypatch,
        facade,
        measured_data_setup,
        valid_obs_data,
    ):
        measured_data_setup(input_dataframe, valid_obs_data, monkeypatch)
        md = MeasuredData(facade, ["obs_key"])
    
        md.remove_failed_realizations()
    
        expected_result.columns = _set_multiindex(expected_result)
        expected_result = pd.concat({"obs_key": expected_result}, axis=1)
>       assert md.data.equals(expected_result)
E       assert False
E        +  where False = <bound method NDFrame.equals of            obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0>(           obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0)
E        +    where <bound method NDFrame.equals of            obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0> =            obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0.equals
E        +      where            obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0 = <ert_data.measured.MeasuredData object at 0x7f148ee2dbe0>.data

tests/ert_tests/data/test_measured_data.py:108: AssertionError
_______________________ test_remove_failed_realizations[input_dataframe2-expected_result2] _______________________

input_dataframe = key_index   0  1     2
data_index  0  1     2
1           7  8  None
expected_result =            obs_key          
key_index        0    1    2
data_index       0    1    2
OBS            1.0  2.0  3.0
STD            4.0  5.0  6.0
1              7.0  8.0  NaN
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f148edd11c0>, facade = <Mock id='139726272135072'>
measured_data_setup = <function measured_data_setup.<locals>._setup at 0x7f148ee5de50>
valid_obs_data = key_index     0    1    2
data_index    0    1    2
OBS         1.0  2.0  3.0
STD         4.0  5.0  6.0

    @pytest.mark.parametrize(
        "input_dataframe,expected_result",
        [
            (
                pd.DataFrame(data=[[7, 8, 9]], index=[1]),
                pd.DataFrame(
                    data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]],
                    index=["OBS", "STD", 1],
                ),
            ),
            (
                pd.DataFrame(data=[[None, None, None]], index=[1]),
                pd.DataFrame(data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]], index=["OBS", "STD"]),
            ),
            (
                pd.DataFrame(data=[[7, 8, None]], index=[1]),
                pd.DataFrame(
                    data=[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, None]],
                    index=["OBS", "STD", 1],
                ),
            ),
        ],
    )
    def test_remove_failed_realizations(
        input_dataframe,
        expected_result,
        monkeypatch,
        facade,
        measured_data_setup,
        valid_obs_data,
    ):
        measured_data_setup(input_dataframe, valid_obs_data, monkeypatch)
        md = MeasuredData(facade, ["obs_key"])
    
        md.remove_failed_realizations()
    
        expected_result.columns = _set_multiindex(expected_result)
        expected_result = pd.concat({"obs_key": expected_result}, axis=1)
>       assert md.data.equals(expected_result)
E       assert False
E        +  where False = <bound method NDFrame.equals of            obs_key           \nkey_index        0    1     2\ndata_index       0    1     2\nOBS            1.0  2.0   3.0\nSTD            4.0  5.0   6.0\n1              7.0  8.0  None>(           obs_key          \nkey_index        0    1    2\ndata_index       0    1    2\nOBS            1.0  2.0  3.0\nSTD            4.0  5.0  6.0\n1              7.0  8.0  NaN)
E        +    where <bound method NDFrame.equals of            obs_key           \nkey_index        0    1     2\ndata_index       0    1     2\nOBS            1.0  2.0   3.0\nSTD            4.0  5.0   6.0\n1              7.0  8.0  None> =            obs_key           \nkey_index        0    1     2\ndata_index       0    1     2\nOBS            1.0  2.0   3.0\nSTD            4.0  5.0   6.0\n1              7.0  8.0  None.equals
E        +      where            obs_key           \nkey_index        0    1     2\ndata_index       0    1     2\nOBS            1.0  2.0   3.0\nSTD            4.0  5.0   6.0\n1              7.0  8.0  None = <ert_data.measured.MeasuredData object at 0x7f148ed0f9a0>.data

tests/ert_tests/data/test_measured_data.py:108: AssertionError
================================================ warnings summary ================================================
../../venv/fu/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191
  /home/thannant/development/equinor/venv/fu/lib/python3.9/site-packages/pytest_asyncio/plugin.py:191: DeprecationWarning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)

tests/ert_tests/data/test_measured_data.py::test_get_data[GEN_OBS]
  /home/thannant/development/equinor/venv/fu/lib/python3.9/site-packages/pytest_asyncio/plugin.py:317: DeprecationWarning: '@pytest.fixture' is applied to <fixture env_save, file=/home/thannant/development/equinor/monkey/fubar/tests/conftest.py, line=23> in 'legacy' mode, please replace it with '@pytest_asyncio.fixture' as a preparation for switching to 'strict' mode (or use 'auto' mode to seamlessly handle all these fixtures as asyncio-driven).
    warnings.warn(

tests/ert_tests/data/test_measured_data.py::test_get_data[GEN_OBS]
  /home/thannant/development/equinor/venv/fu/lib/python3.9/site-packages/pytest_asyncio/plugin.py:317: DeprecationWarning: '@pytest.fixture' is applied to <fixture monkeypatch, file=/home/thannant/development/equinor/venv/fu/lib/python3.9/site-packages/_pytest/monkeypatch.py, line=29> in 'legacy' mode, please replace it with '@pytest_asyncio.fixture' as a preparation for switching to 'strict' mode (or use 'auto' mode to seamlessly handle all these fixtures as asyncio-driven).
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================ short test summary info =============================================
FAILED tests/ert_tests/data/test_measured_data.py::test_remove_failed_realizations[input_dataframe1-expected_result1]
FAILED tests/ert_tests/data/test_measured_data.py::test_remove_failed_realizations[input_dataframe2-expected_result2]
 `


**Enviromment**
 - OS: ubuntu/debian/macos
 - Python version 3.8/3.9

--- a/setup.py
+++ b/setup.py
@@ -87,7 +87,7 @@ setup(
         "matplotlib",
         "numpy",
         "packaging",
-        "pandas",
+        "pandas=<1.4.0",
         "pluggy",
         "prefect",
         "psutil",

--- a/setup.py
+++ b/setup.py
@@ -87,7 +87,7 @@ setup(
         "matplotlib",
         "numpy",
         "packaging",
-        "pandas=<1.4.0",
+        "pandas <= 1.3.5",
         "pluggy",
         "prefect",
         "psutil",

